src/Exception/BlockServiceNotFoundException.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\BlockBundle\Exception;
+
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class BlockServiceNotFoundException extends NotFoundHttpException
+{
```

Comment:
Should it extend from http exception 🤔

If yes then we should add NotFoundHttpException

Reply:
Maybe not, I copied the old exception. And what about that ?
```php
namespace Sonata\BlockBundle\Exception;

final class BlockServiceNotFoundException implements BlockExceptionInterface
{
}
```
Or simply :
```php
final class BlockServiceNotFoundException
{
```
Let me know and I'll do modifications tomorrow morning

Reply:
it needs to extend some kind of exception. If you also implement the BlockExceptionInterface maybe thats enough to render empty response automatically? or not?

I feel like the whole exception catching should happen in the block bundle, are you sure you want to introduce specific logic for page @VincentLanglet ?

Reply:
> I feel like the whole exception catching should happen in the block bundle, are you sure you want to introduce specific logic for page @VincentLanglet ?

do you mean here https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Templating/Helper/BlockHelper.php#L388 @jordisala1991 ?

But if we do this, we can't handle the exception for code that uses this method, if for some reason I need to implement some logical if the block doesn't exist, I can't do that, because I never get the exception!

usually I handle the exception in the last layer before to return to client.

Reply:
First I dunno why BlockNotFound extends NotFoundException, it feels weird https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Exception/BlockNotFoundException.php#L18

I mean, I don't see how we're in a Http context here: https://github.com/sonata-project/SonataBlockBundle/blob/375b168da31545bf3c386fd25a4b483411ac97c5/src/Block/BlockLoaderChain.php#L66 and we should throw a 404.

And indeed there is a BlockExceptionInterface.

Seems like a strategy exist to ignore block exceptions https://github.com/sonata-project/SonataBlockBundle/blob/987d2936d3d2a3d4fe4615241f545aad693eb4c2/docs/reference/exceptions.rst#filters

So indeed, throwing a BlockServiceNotFound extends BlockExceptionInterface would allow people to ignore this exception with the correct filter.
It might require some conf in PageBundle still to use the correct filter.

But again, I asked a question and got no answer so far:
which filter and which renderer is using SonataPageBundle.
https://github.com/sonata-project/SonataBlockBundle/blob/987d2936d3d2a3d4fe4615241f545aad693eb4c2/docs/reference/exceptions.rst#exception-strategy

IMHO, blockbundle should allow to ignore them (with specific exception/config) and pagebundle should activate this ; and this already a feature from blockbundle.

Reply:
I didn't know this exception strategy in block bundle 👀 
Well if we return an exception that implements this interface we just need to enable it into the configs! we can  let this handle as true for default block config :)

Reply:
we have too many ways to handle it 😄

Reply:
So to go forward I have to remove old `extend NotFoundHttpException` and add `implements BlockExceptionInterface ` in both files src/Exception/BlockServiceNotFoundException.php and src/Exception/BlockNotFoundException.php ?

```php
namespace Sonata\BlockBundle\Exception;
```

```php
final class BlockServiceNotFoundException implements BlockExceptionInterface
{
}
```
```php
final class BlockNotFoundException implements BlockExceptionInterface
{
}
```

And change configs into page bundle with [exception strategy feature](https://github.com/sonata-project/SonataBlockBundle/blob/987d2936d3d2a3d4fe4615241f545aad693eb4c2/docs/reference/exceptions.rst#exception-strategy). @eerison do you look for this part ?

Reply:
I'm not sure for BlockNotFoundException, technically it would be a BC break

Reply:
So:
- no changes for `BlockNotFoundException.php`
- and only add implements `BlockExceptionInterface` for `BlockServiceNotFoundException.php`

Reply:
> I'm not sure for BlockNotFoundException, technically it would be a BC break

It's a good example to change just what is really necessary 😄

Reply:
@JeromeEngelnAdeliom  could you change only `BlockServiceNotFoundException` please!

and other thing I guess you need to extend from some exception, right @VincentLanglet ?

in this case which exception? `Exception` or `NotFoundHttpException`?

Reply:
It would also be nice if someone checks why that interface was introduced and why they didn't implement any exception with that interface, maybe there is a reason, maybe its just one more "WTF".

Reply:
After @JeromeEngelnAdeliom  change it, I'll test in sonata page bundle, and check if those configs works as expected :D

Reply:
> It would also be nice if someone checks why that interface was introduced and why they didn't implement any exception with that interface, maybe there is a reason, maybe its just one more "WTF".

Ten years ago https://github.com/sonata-project/SonataBlockBundle/pull/36 :(

Reply:
@eerison  wdyt about using `extends \UnexpectedValueException` ? it seems to match with our missing block service case.
=> Exception thrown if a value does not match with a set of values. Typically this happens when a function calls another function and expects the return value to be of a certain type or value not including arithmetic or buffer related errors.

Reply:
IMO `BlockServiceNotFoundException` isn't a abstraction of `UnexpectedValueException` 👀

Reply:
Extending Exception is enough.

And fun fact, @jordisala1991, https://github.com/sonata-project/SonataBlockBundle/blob/4.x/src/Exception/BlockExceptionInterface.php#L19 is currently not extending Throwable...